sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.exception.AzureException;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.rest.Response;
+import com.azure.core.implementation.TypeUtil;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.polling.implementation.PollingConstants;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+/**
+ * Implements a Location polling strategy.
+ *
+ * @param <T> the type of the response type from a polling call, or BinaryData if raw response body should be kept
+ * @param <U> the type of the final result object to deserialize into, or BinaryData if raw response body should be
+ *           kept
+ */
+public class LocationPollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private static final SerializerAdapter SERIALIZER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    private final HttpPipeline httpPipeline;
+    private final Context context;
+
+    /**
+     * Creates an instance of the location polling strategy.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public LocationPollingStrategy(
+            HttpPipeline httpPipeline,
+            Context context) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        HttpHeader locationHeader = initialResponse.getHeaders().get(PollingConstants.LOCATION);
+        return Mono.just(locationHeader != null);
+    }
```

Comment:
Not a blocking issue.

You might want to double check whether the header is indeed an URI. We had issue of invalid `Azure-AsyncOperation` header (which is defined for MS internal use).
https://github.com/Azure/azure-sdk-for-java/issues/22032#issuecomment-855588700

And `Location` header would be more likely to be used for things other than LRO (redirection, etc.).

Reply:
> And Location header would be more likely to be used for things other than LRO (redirection, etc.).

I think having it close to the end of the chain will reduce the chance of misusing it from its original intention. If an API is labeled `x-ms-long-running-operation` and returns `Location` header for other things other than LRO and uses another way of handling LRO, a custom polling strategy is probably needed for it.

Reply:
I also added check of the headers - if it's not a URL `canPoll` will return false

Reply:
It is OK. Just that `URL` in Java actually do not check much. Hence our problem in the issue mentioned above (URL check pass, but still contains space). https://stackoverflow.com/a/5965755